File: labs1/5hw_ariphmetic_parser/expression/exceptions/ParsingException.java
package expression.exceptions;

public class ParsingException extends Exception {
    public ParsingException(final String message) {
        super(message);
    }

    static protected String getPlace(final int ind, final int size) {
        StringBuilder res = new StringBuilder();
        for (int i = 0; i < ind; i++) {
            res.append('~');
        }
        res.append("^\n");
        return res.toString() + Colors.RESET;
    }
}

File: android/Hw1_calculator/app/src/main/java/com/lab/hw1/parser/exceptions/ParsingException.kt
package com.lab.hw1.parser.exceptions

import java.lang.RuntimeException

open class ParsingException(message: String, ind: Int) : RuntimeException("\n$message\n${getPlace(ind)}") {

}

private fun getPlace(ind: Int): String {
    val res = StringBuilder()
    for (i in 0 until ind + 1) {
        res.append('~')
    }
    res.append("^\n")
    return res.toString()
}
